internal/server/v1/propertyvalues/heap.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package propertyvalues
+
+import (
+	pb "github.com/datacommonsorg/mixer/internal/proto"
+)
+
+type heapElem struct {
+	ig   int
+	pos  int32
+	data *pb.EntityInfo
+}
+
+// An entityHeap is a min-heap of entity.
+type entityHeap []*heapElem
+
+func (h entityHeap) Len() int { return len(h) }
+func (h entityHeap) Less(i, j int) bool {
+	di := h[i].data
+	dj := h[j].data
+	// This needs to match the sorting in Prophet.
```

Comment:
Can you elaborate more?  We can also document this in the place where the key is defined for these caches.